core/src/main/java/nu/yona/server/exceptions/SignInValidationException.java

```diff
@@ -0,0 +1,34 @@
+package nu.yona.server.exceptions;
+
+/**
+ * This exception is to be used in case user sign-in code is wrong.
+ */
+public class SignInValidationException extends YonaException
+{
+	private static final long serialVersionUID = -7917208280838423613L;
+
+	private SignInValidationException(String messageId, Object... parameters)
+	{
+		super(messageId, parameters);
+	}
+
+	private SignInValidationException(Throwable t, String messageId, Object... parameters)
+	{
+		super(t, messageId, parameters);
+	}
+
+	public static SignInValidationException signInCodeMismatch()
+	{
+		return new SignInValidationException("error.sign-in.code.mismatch");
```

Comment:
Rename into ``error.mobile.number.confirmation.code.mismatch`` and use a similar pattern for the other resource IDs.